File: Python/Algorithms/2042.py
class Solution:
    def areNumbersAscending(self, s: str) -> bool:
        prev, next_ = 0, 0
        start, end = 0, len(s)
        while start < end:
            digits = []
            while start < end and s[start].isdigit():
                digits.append(s[start])
                start += 1
            if digits:
                next_ = int("".join(digits))
                if next_ <= prev:
                    return False
                prev = next_
            start += 1
        return True

File: Rust/Algorithms/2042.rs
impl Solution {
    pub fn are_numbers_ascending(s: String) -> bool {
        let (mut previous, mut next) = (0, 0);
        let mut it = s.chars();
        loop {
            let mut c = match it.next() {
                None => break,
                Some(c) => c,
            };
            let mut digit = 0;
            loop {
                if c.is_digit(10) {
                    digit *= 10;
                    digit += c.to_digit(10).unwrap();
                    c = match it.next() {
                        None => break,
                        Some(c) => c,
                    };
                } else {
                    break;
                }
            }
            if digit > 0 {
                next = digit;
                if next <= previous {
                    return false;
                }
                previous = next;
            }
        }
        return true;
    }
}